sdk/communication/communication-phone-numbers/test/public/utils/operationLocationFixPolicy.ts

```diff
@@ -0,0 +1,140 @@
+          const queryParams = urlObj.searchParams.toString();
+
+          // Common LRO result endpoints for Azure Communication Services
+          fixedUrl = `${SANITIZED_BASE_URL}/availablePhoneNumbers/searchResults/00000000-0000-0000-0000-000000000000${queryParams ? `?${queryParams}` : ""}`;
+
+          // eslint-disable-next-line no-param-reassign
+          request.url = fixedUrl;
+        }
+
+        // Pattern 2: URLs pointing to example.com
+        else if (originalUrl.includes("example.com")) {
+          const urlObj = new URL(originalUrl);
+          const queryParams = urlObj.searchParams.toString();
+
+          // Try to infer the correct path based on common patterns
+          if (queryParams.includes("api-version")) {
+            // Default to phoneNumbers operations path if we can't determine the specific operation
+            fixedUrl = `${SANITIZED_BASE_URL}/phoneNumbers/operations/search_00000000-0000-0000-0000-000000000000${queryParams ? `?${queryParams}` : ""}`;
+          }
+
+          // eslint-disable-next-line no-param-reassign
+          request.url = fixedUrl;
+        }
+      }
+
+      const response = await next(request);
+
+      // Fix response headers (only during playback)
+      if (isPlaybackMode() && response.headers) {
+        const apiVersion = request.url ? getApiVersion(request.url) : "2025-04-01";
+        const operationLocation =
+          response.headers.get("operation-location") || response.headers.get("Operation-Location");
+        const operationId = response.headers.get("operation-id");
+        const location = response.headers.get("location") || response.headers.get("Location");
+
+        // Fix Operation-Location header if it's broken
+        if (
+          operationLocation &&
+          (operationLocation === "https://example.com" ||
+            operationLocation === SANITIZED_BASE_URL ||
+            operationLocation.includes("example.com") ||
+            operationLocation.endsWith("/Sanitized"))
+        ) {
+          let fixedOperationUrl: string;
+
+          if (operationId) {
+            // Use the operation ID to construct the correct URL
+
+            // Determine operation type from operation ID pattern
+            if (operationId.startsWith("search_")) {
+              fixedOperationUrl = `${SANITIZED_BASE_URL}/phoneNumbers/operations/${operationId}?api-version=${apiVersion}`;
+            } else if (operationId.startsWith("purchase_")) {
+              fixedOperationUrl = `${SANITIZED_BASE_URL}/phoneNumbers/operations/${operationId}?api-version=${apiVersion}`;
+            } else if (operationId.startsWith("release_")) {
+              fixedOperationUrl = `${SANITIZED_BASE_URL}/phoneNumbers/operations/${operationId}?api-version=${apiVersion}`;
+            } else {
+              // Generic fallback for any other operation type
+              fixedOperationUrl = `${SANITIZED_BASE_URL}/phoneNumbers/operations/${operationId}?api-version=${apiVersion}`;
```

Comment:
Why are all of these if/else statements the same? The value of `fixedOperationUrl` is always the same, regardless of the value of `operationId`